WordPress/src/main/java/org/wordpress/android/ui/prefs/AppSettingsFragment.java

```diff
@@ -134,23 +134,25 @@ private void changeLanguage(String languageCode) {
 
         Resources res = getResources();
         Configuration conf = res.getConfiguration();
-        Locale currentLocale = conf.locale != null ? conf.locale : LanguageUtils.getCurrentDeviceLanguage(WordPress.getContext());
+        Locale currentLocale = conf.locale != null ?
+                conf.locale : LanguageUtils.getCurrentDeviceLanguage(WordPress.getContext());
 
-        if (currentLocale.toString().equals(languageCode)) return;
-
-        updateLanguagePreference(languageCode);
+        if (currentLocale.getLanguage().equals(new Locale(languageCode).getLanguage())) {
```

Comment:
This is how [the documentation](https://docs.oracle.com/javase/7/docs/api/java/util/Locale.html#getLanguage()) for `getLanguage` recommends comparing languages.